src/components/vice/admin/table/index.js

```diff
@@ -17,79 +17,181 @@ import {
+import useStyles from "./styles";
 
 // Constructs an ID for an element.
 const id = (...names) => buildID(ids.BASE, ...names);
 
-const useStyles = makeStyles((theme) => ({
-    root: {
-        width: "100%",
-    },
-    paper: {
-        width: "100%",
-        marginBottom: theme.spacing(5),
-    },
-    title: {
-        padding: theme.spacing(2),
-    },
-    table: {
-        height: "100%",
-    },
-    extended: {
-        display: "flex",
-        marginLeft: theme.spacing(7),
-        marginRight: theme.spacing(7),
-
-        [theme.breakpoints.down("sm")]: {
-            marginLeft: theme.spacing(1),
-            marginRight: theme.spacing(1),
-        },
-
-        flexWrap: "wrap",
-        flexShrink: 0,
-        flexGrow: 0,
-    },
-    row: {
-        "& > *": {
-            borderBottom: "unset",
-        },
-    },
-    dataEntry: {
-        [theme.breakpoints.up("xs")]: {
-            width: "100%",
-            marginLeft: 0,
-            marginRight: 0,
-            marginTop: theme.spacing(1),
-            marginBottom: theme.spacing(1),
-        },
-        [theme.breakpoints.up("sm")]: {
-            width: 300,
-            marginLeft: 0,
-            marginRight: 0,
-            marginTop: theme.spacing(1),
-            marginBottom: theme.spacing(1),
-        },
-        [theme.breakpoints.up("lg")]: {
-            width: 350,
-            margin: theme.spacing(2),
-        },
-    },
-    dataEntryLabel: {
-        marginRight: theme.spacing(1),
-        fontWeight: 500,
-    },
-}));
-
-const ExtendedDataCard = ({ columns, row, collapseID }) => {
+const ActionButtonsSkeleton = () => {
+    return (
+        <>
+            <Skeleton
+                variant="rect"
+                animation="wave"
+                height={75}
+                width="100%"
+            />
+        </>
+    );
+};
+
+const ActionButton = ({ baseID, name, handler, onClick, popperMsgKey }) => {
+    const classes = useStyles();
+    return (
+        <Button
+            id={id(baseID, "button", name)}
+            variant="contained"
+            color="primary"
+            onClick={(event) => onClick(event, handler, popperMsgKey)}
+            className={classes.actionButton}
+        >
+            {msg(name)}
+        </Button>
+    );
+};
+
+const ActionButtons = ({
+    row,
+    handleExtendTimeLimit = (_) => {},
+    handleDownloadInputs = (_) => {},
+    handleUploadOutputs = (_) => {},
+    handleExit = (_) => {},
+    handleSaveAndExit = (_) => {},
+}) => {
+    const classes = useStyles();
+
+    const [anchorEl, setAnchorEl] = useState(null);
+    const [open, setOpen] = useState(false);
+    const [popperMessage, setPopperMessage] = useState("");
+
+    const { status, data, error } = useQuery(
+        ["async-data", row.externalID],
+        asyncData
+    );
+
+    const isLoading = status === "loading";
+    const hasErrored = status === "error";
+
+    if (hasErrored) {
+        console.log(error);
```

Comment:
We should probably add in the same error handling we have in other components, using components from `/utils/error`

Reply:
I'm going to be working on another set of features for the admin page soon, I'll include this in those PRs.